packages/block-editor/src/components/block-list/index.native.js

```diff
@@ -149,6 +152,27 @@ export class BlockList extends Component {
 		);
 	}
 
+	cellRenderer( { children, index, item } ) {
+		const { shouldAllowSpaceForFloatingToolbar } = this.props;
+		const adjustsForFloatingToolbar = shouldAllowSpaceForFloatingToolbar(
+			item
+		);
+
+		return (
+			<View
+				zIndex={ 100 + index }
```

Comment:
In case of inner blocks there will be cases where "index" of the parent block is equal or bigger than the child block, could this cause something odd? I don't have any real scenario in mind but it feels odd to have bigger z-index on shallow levels compared to deeper levels. Maybe it is safer to update this as: 100*depth + index

Reply:
> Maybe it is safer to update this as: 100*depth + index

That's a good idea. I haven't put a lot of thought into how I would handle this long term, but I like that suggestion. I'll work that into the PR.